### PR TITLE
cluster: definition v1.8

### PR DESCRIPTION
**Pull request by pinebit**
Introducing cluster definition v1.8 with partial deposits.

category: feature
ticket: none

### CODE REVIEW

File: cluster/testdata/cluster_lock_v1_0_0.json [hunk 1]
```diff
@@ -38,13 +38,13 @@
    ]
   },
   {
-   "distributed_public_key": "0xf2468928d5a23b9ca740f80c9382d9c6034ad2960c796503e1ce221725f50caf1fbfe831b10b7bf5b15c47a53dbf8e7d",
```
Comment (pinebit) [inline cluster/testdata/cluster_lock_v1_0_0.json:41]: @xenowits is it normal that these locks were updated during go generate?

**Review by gsora — APPROVED** [review]

File: cluster/testdata/cluster_lock_v1_8_0.json [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+   "public_shares": [
+    "0x975deda77e758579ea3dfe4136abf752b3b8271d03e944b3c9db366b75045f8efd69d22ae5411947cb553d7694267aef",
+    "0x4ebcea406b32d6108bd68584f57e37caac6e33feaa3263a399437024ba9c9b14678a274f01a910ae295f6efbfe5f5abf"
+   ],
+   "deposit_data": {
+    "pubkey": "",
+    "withdrawal_credentials": "",
+    "amount": "0",
+    "signature": ""
+   },
```
Comment (dB2510) [inline cluster/testdata/cluster_lock_v1_8_0.json:58]: This shouldn't be in v1.8.0

File: cluster/testdata/cluster_lock_v1_8_0.json [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+     "timestamp": 1655733600,
+     "pubkey": "0x1814be823350eab13935f31d84484517e924aef78ae151c00755925836b7075885650c30ec29a3703934bf50a28da102"
+    },
+    "signature": "0xd313c8a3b4c1c0e05447f4ba370eb36dbcfdec90b302dcdc3b9ef522e2a6f1ed0afec1f8e20faabedf6b162e717d3a748a58677a0c56348f8921a266b11d0f334c62fe52ba53af19779cb2948b6570ffa0b773963c130ad797ddeafe4e3ad29b"
+   },
+   "partial_deposit_data": [
+    {
+     "pubkey": "0x1814be823350eab13935f31d84484517e924aef78ae151c00755925836b7075885650c30ec29a3703934bf50a28da102",
+     "withdrawal_credentials": "0x76b0620556304a3e3eae14c28d0cea39d2901a52720da85ca1e4b38eaf3f44c6",
+     "amount": "5919415281453547599",
```
Comment (dB2510) [inline cluster/testdata/cluster_lock_v1_8_0.json:72]: this amount doesn't look right?

**Review by dB2510 — COMMENTED** [review]
Please the the testdata/ json artefacts of cluster lock. They are not reflecting the logic properly.

File: cluster/testdata/cluster_lock_v1_8_0.json (hunk 3 above)
Comment (pinebit) [reply cluster/testdata/cluster_lock_v1_8_0.json:72]: This is randomly generated amount. It does not have to be 32ETH for example for the purpose of the test.

File: cmd/createcluster_internal_test.go [hunk 4]
```diff
@@ -325,7 +325,7 @@ func testCreateCluster(t *testing.T, conf clusterConfig, def cluster.Definition,
 			}
 
 			if isAnyVersion(lock.Version, "v1.6.0", "v1.7.0") {
-				require.NotEmpty(t, val.DepositData)
+				require.NotEmpty(t, val.PartialDepositData)
```
[the remark this answers is not in the document]
Comment (pinebit) [reply cmd/createcluster_internal_test.go:328]: Good point!

File: cluster/testdata/cluster_lock_v1_8_0.json (hunk 2 above)
Comment (pinebit) [reply cluster/testdata/cluster_lock_v1_8_0.json:58]: You are very right! My bad, now fixed.

File: cluster/distvalidator.go [hunk 5]
```diff
@@ -96,21 +96,30 @@ type distValidatorJSONv1x2to5 struct {
...
 	PubShares           []ethHex                `json:"public_shares,omitempty"`
 	DepositData         depositDataJSON         `json:"deposit_data,omitempty"`
 	BuilderRegistration builderRegistrationJSON `json:"builder_registration,omitempty"`
 }
 
+// distValidatorJSONv1x8 is the json formatter of DistValidator for versions v1.8.0 or later.
+type distValidatorJSONv1x8 struct {
+	PubKey              ethHex                  `json:"distributed_public_key"`
+	PubShares           []ethHex                `json:"public_shares,omitempty"`
+	DepositData         depositDataJSON         `json:"deposit_data,omitempty"`
```
[the remark this answers is not in the document]
Comment (pinebit) [reply cluster/distvalidator.go]: You are correct, removed.

File: cluster/definition.go [hunk 6]
```diff
@@ -685,6 +723,37 @@ func unmarshalDefinitionV1x5(data []byte) (def Definition, err error) {
 	}, nil
 }
 
+func unmarshalDefinitionV1x8(data []byte) (def Definition, err error) {
```
[the remark this answers is not in the document]
Comment (pinebit) [reply cluster/definition.go:726]: I am going to revisit verification end-to-end in the "final" integrating PR. The codebase of this PR does not have the verification method accessible yet.

**Review by dB2510 — APPROVED** [review]
LGTM

**Pull request merged**